### PR TITLE
Feature/visual tests

### PR DESCRIPTION
**Pull request by villenavg**

### CODE REVIEW

File: OWDTestToolkit/utils/engines/perceptualdiff_engine.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    perceptualdiff_output_png = True
+
+    def assertSameFiles(self, output_file, baseline_file, threshold):
+        # Calculate threshold value as a pixel number instead of percentage.
+        width, height = Image.open(output_file).size
+        threshold = int(width * height * threshold)
+
+        diff_ppm = output_file.replace(".png", ".diff.ppm")
+        cmd = "%s -threshold %d -output %s %s %s" % (
+            self.perceptualdiff_path, threshold, diff_ppm, baseline_file, output_file)
```
Comment (FranciscoCano) [inline OWDTestToolkit/utils/engines/perceptualdiff_engine.py]: uhmmmmm... %...

File: OWDTestToolkit/utils/engines/perceptualdiff_engine.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        else:
+            if os.path.exists(diff_ppm):
+                if self.perceptualdiff_output_png:
+                    # Convert the .ppm output to .png
+                    diff_png = diff_ppm.replace("diff.ppm", "diff.png")
+                    Image.open(diff_ppm).save(diff_png)
+                    os.remove(diff_ppm)
+                    diff_file_msg = ' (See %s)' % diff_png
+                else:
+                    diff_file_msg = ' (See %s)' % diff_ppm
```
Comment (FranciscoCano) [inline OWDTestToolkit/utils/engines/perceptualdiff_engine.py]: %

File: OWDTestToolkit/utils/engines/perceptualdiff_engine.py (hunk 1 above)
Comment (villenavg) [inline OWDTestToolkit/utils/engines/perceptualdiff_engine.py]: Don't look at me... It was like that when I arrived

File: config/gaiatest_testvars.json [hunk 3]
```diff
@@ -101,6 +101,13 @@
 		"device_cert": "cert_device",
 		"branch_cert": "v2.0",
 		"buildname_cert": "cert_build-.master.B-xxx.Gecko-xxxxxxx.Gaia-xxxxxxx"
+	},
+	"visualtest": {
+		"enabled": true,
+		"engine": "pil",
+		"save_baseline": false,
+		"baseline_dir": "/home/avillena/OWD/owd_test_cases/screenshots/baseline",
+		"cmp_dir": "/home/avillena/OWD/owd_test_cases/screenshots/cmp"
```
Comment (FranciscoCano) [inline config/gaiatest_testvars.json]: These paths should exist in CI

File: OWDTestToolkit/utils/engines/perceptualdiff_engine.py (hunk 1 above)
Comment (FranciscoCano) [inline OWDTestToolkit/utils/engines/perceptualdiff_engine.py]: Understood... ;)

File: config/gaiatest_testvars.json [hunk 4]
```diff
@@ -101,6 +101,13 @@
 		"device_cert": "cert_device",
 		"branch_cert": "v2.0",
 		"buildname_cert": "cert_build-.master.B-xxx.Gecko-xxxxxxx.Gaia-xxxxxxx"
+	},
+	"visualtest": {
+		"enabled": true,
+		"engine": "pil",
+		"save_baseline": false,
+		"baseline_dir": "",
+		"cmp_dir": ""
```
Comment (FranciscoCano) [inline config/gaiatest_testvars.json]: Don't forget to update .boe and .mad files, so they are copied to their respective machines

Comment (villenavg) [inline config/gaiatest_testvars.json]: I wanted to discuss something first with you, before updating them. There will be a second PR to deal with it

Comment (villenavg) [inline config/gaiatest_testvars.json]: :sunglasses:

Comment (villenavg) [inline config/gaiatest_testvars.json]: Nevermind

Comment (FranciscoCano) [inline config/gaiatest_testvars.json]: Waaaat? :dancer:

File: config/gaiatest_testvars.json.boe [hunk 5]
```diff
@@ -100,5 +100,12 @@
 		"device_cert": "cert_device",
 		"branch_cert": "v2.0",
 		"buildname_cert": "cert_build-.master.B-xxx.Gecko-xxxxxxx.Gaia-xxxxxxx"
+	},
+	"visualtest": {
+		"enabled": true,
+		"engine": "pil",
+		"save_baseline": false,
+		"baseline_dir": "/home/develenv/projects/OWD/owd_test_cases/screenshots/baseline",
```
Comment (FranciscoCano) [inline config/gaiatest_testvars.json.boe]: no OWD part in path :D

File: config/gaiatest_testvars.json (hunk 4 above)
Comment (villenavg) [inline config/gaiatest_testvars.json]: Nothing, "pajas mentales mías". Files already updated :)

**Comment by FranciscoCano** [comment]
LGTM

**Pull request merged**